src/mock-doc/window.ts

```diff
@@ -200,6 +201,20 @@ export class MockWindow {
     return ElementCstr;
   }
 
+  get NodeList() {
+    let NodeListCstr = nodeListCstrMap.get(this);
+    if (NodeListCstr == null) {
+      const ownerDocument = this.document;
+      NodeListCstr = class extends MockElement {
```

Comment:
This can't be right. should not it be a NodeList mock?

Reply:
Cheers for the feedback. I am not sure I fully understand how you guys are mocking each part of the DOM API but I totally misunderstood what Ken did in his PR. I have pushed another commit where I have added a MockNodeList class as this seems to be the pattern you use.

If I understand you are mapping the class members to the available properties of the DOM API objects so then when those get called in a test code (in a mock window context) they do not throw Errors ? 

Let me know if there are any more issues and I will get to it as soon as I can.

Reply:
Per conversation with Adam and Manu, this can be merged as-is and if we need to make it more complete later we can.